users/middleware.py

```diff
@@ -1,39 +1,17 @@
-from django.utils import timezone
+from users.models import Domain
 
-from users.models import Identity, User
 
-
-class IdentityMiddleware:
+class DomainMiddleware:
     """
-    Adds a request.identity object which is either the current session's
-    identity, or None if they have not picked one yet/it's invalid.
+    Tries to attach a Domain object to every incoming request, if one matches.
     """
 
     def __init__(self, get_response):
         self.get_response = get_response
 
     def __call__(self, request):
-        # The API middleware might have set identity already
-        if not hasattr(request, "identity"):
-            # See if we have one in the session
-            identity_id = request.session.get("identity_id")
-            if not identity_id:
-                request.identity = None
-            else:
-                # Pull it out of the DB and assign it
-                try:
-                    request.identity = Identity.objects.get(id=identity_id)
-                    User.objects.filter(pk=request.user.pk).update(
-                        last_seen=timezone.now()
-                    )
-                except Identity.DoesNotExist:
-                    request.identity = None
-
+        request.domain = None
+        if "host" in request.headers:
+            request.domain = Domain.get_domain(request.headers["host"])
```

Comment:
My django is a bit rusty, but shouldn't we be caching this in the session object and only reading from the db on a miss? (Or am I misunderstanding how this works)

Reply:
Yeah, it should probably be cached _somewhere_, but it's not a massive hit to do it once per page load.